### ISSUE TITLE
Access to streetcomplete.app is blocked from Russia

### ISSUE THREAD

**Issue by deevroman** [streetcomplete/StreetComplete#6040]
The app's website, as well as resources such as hosting photos are no longer available in notes from Russian IP addresses (HTTP 403)

https://check-host.net/check-report/215bb21ckae9

Is this an intentional block or a consequence of, for example, DDoS protection? (since I see a blocked IP from India)

**Comment by matkoniecz** [comment]
It works from Poland. Is it blocked by Russian censorship by any chance? (likely as collateral damage, I do not expect it to be important/relevant enough to be blocked deliberately)

**Comment by deevroman** [comment]
Access to the site is available, but the server returns HTTP 403. In other words, this is a server-side restriction

**Comment by matkoniecz** [comment]
Hmm, I expected that connection is hijacked and fake Russian government server returns 403. But I am not familiar with details how they censor on technical site.

**Comment by deevroman** [comment]
Luckily, they can't bypass HTTPS undetected :) Prohibited HTTPS traffic is simply blocked.

**Comment by mnalis** [comment]
@deevroman can you click on padlock icon in browser, and click on "view certificate" to give us SHA-256 fingerprint of the `streetcomplete.app` TLS certificate? Also, when you type `ping streetcomplete.app`, what IP address does it resolve to?

(for me in Croatia, it is on `62.108.32.100` and SHA256 is `82:6D:1D:BF:5B:23:28:E5:3D:8C:8B:D7:87:C8:41:64:27:F0:11:65:56:B0:4D:97:F6:AD:10:B9:1D:71:6F:27`, and works fine).

> Is this an intentional block or a consequence of, for example, DDoS protection? (since I see a blocked IP from India)

Hmm, possible, but if it is, it seems to overreaching (as there is not just one but all of the Russian testpoints that got blocked according to that report of your) (yeah things sometimes do get blocked in such ways as a response to DDoS, but they are usually at layer3 level (not announcing BGP IP prefixes or droping whole ASes) and not on layer7 (HTTP 403).  
For 
shared hosts, more probable cause might be government mandated blockade or ISP-level blocking decision as temporary stopgap response for some app-level vulnerability, but for VPSs where the owner runs its own webserver (which I thinks is the case here) it shouldn't be anyone elses decision but the VPS operators (direct or indirect via RBL blacklists etc) (again, unless that IP/certificate is MiTM'd)

**Comment by deevroman** [comment]
If it was a MiTM, the browser would have warned me. By the way, HTTP 403 is returned not only for me, but also for other cartographers from Russia.

<img width="763" src="https://github.com/user-attachments/assets/c1bd7c07-ab95-4903-bf45-bafda05e310e">

<details>
<summary>curl -v https://streetcomplete.app</summary>

* Host streetcomplete.app:443 was resolved.
* IPv6: (none)
* IPv4: 62.108.32.100
*   Trying 62.108.32.100:443...
* Connected to streetcomplete.app (62.108.32.100) port 443
* ALPN: curl offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
* (304) (IN), TLS handshake, Unknown (8):
* (304) (IN), TLS handshake, Certificate (11):
* (304) (IN), TLS handshake, CERT verify (15):
* (304) (IN), TLS handshake, Finished (20):
* (304) (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / AEAD-CHACHA20-POLY1305-SHA256 / [blank] / UNDEF
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=streetcomplete.app
*  start date: Sep 28 16:18:08 2024 GMT
*  expire date: Dec 27 16:18:07 2024 GMT
*  subjectAltName: host "streetcomplete.app" matched cert's "streetcomplete.app"
*  issuer: C=US; O=Let's Encrypt; CN=R10
*  SSL certificate verify ok.
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://streetcomplete.app/
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: streetcomplete.app]
* [HTTP/2] [1] [:path: /]
* [HTTP/2] [1] [user-agent: curl/8.7.1]
* [HTTP/2] [1] [accept: */*]
> GET / HTTP/2
> Host: streetcomplete.app
> User-Agent: curl/8.7.1
> Accept: */*
>
* Request completely sent off
< HTTP/2 403
< server: nginx
< date: Mon, 09 Dec 2024 15:18:09 GMT
< content-type: text/html
< content-length: 146
< vary: Accept-Encoding
<
<html>
<head><title>403 Forbidden</title></head>
<body>
<center><h1>403 Forbidden</h1></center>
<hr><center>nginx</center>
</body>
</html>
* Connection #0 to host streetcomplete.app left intact

</details>

**Comment by mnalis** [comment]
**TL;DR:** the issue seems to be confirmed to be with the **access control lists of nginx server which runs https://streetcomplete.app website**. As that is the same IP that runs custom-software westnorost.de servers, I'll venture to guess that that is the same VPS under control of @westnordost -- so if it blocks connections from Russia with `403`, I'd wager that that is likely because it contains some (manual or auto-updating[^1]) blocklist(s). 

---

> If it was a MiTM, the browser would have warned me

That for that logs. For regular malware, yes, you are correct. For state-class actors, who are likely able to control CAs that browsers of their citizens use, they have ways to avoid that. But I do not know how deep such control in Russia (on anywhere else) might go :shrug: . Our whole web system of trust (`https://`) is very weak[^2] at _preventing_ such abuses[^3].

Your checks in combination RIPE Atlas measurement I've run https://atlas.ripe.net/measurements/84294392/overview confirm that no such MiTM is taking places. 


[^1]:  it might even be some automated tool (like fail2ban or whatever) which monitors excessive traffic (in which case [429](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) would be better error code) or excessive hacking attempts and blocks whole IP ranges or even ASNs. Or more likely (given whole contries going down at once) some [RBL](https://en.wikipedia.org/wiki/Domain_Name_System_blocklist) or similar blacklist in use declared (by design or by error) those IP as problematic . (Or it could be blocking on something other by IP, for example by prefered language or user-agent, but those are less likely scenarios...)

[^2]: look at your browser CA store -- do you personally trust each and every of those CAs (and theirsubsidearies) not to misuse their power (nor being exploited) under any circumstances? Because _any_ of them can, at _any_ moment fake **_valid_** certificate for _any_ website  (only issue is they'd very likely be detected and get revoked their CA status, so it's not profitable for them to do so -- but their will for continuing to profit is main thread that holds that pretense of security together. Although we have [CAA DNS](https://en.wikipedia.org/wiki/DNS_Certification_Authority_Authorization) records which in combination with DNSSEC might make it somewhat better in the future, adaptation is AFAIK still quite low. Currently if you use those, in only helps with easier detection of the abuse, not in its prevention). But I digress.

[^3]:  but relatively good at detecting them with [CT logs](https://en.wikipedia.org/wiki/Certificate_Transparency)

**Comment by mnalis** [comment]
Ah, that seems to be shared hosting (and not VPS under full control as previously guessed)... 
Still, @westnordost as a customer should probably complain to his web hoster (wint.global / comtrance GmbH?) who seems to be responsible for it, as it blocks both:

- other websites on the same IP:  https://check-host.net/check-report/215d2cd9kd95, _and_
- websites on other, close-by IPs https://check-host.net/check-report/215d27e7kad

(that does not preclude you @deevroman to inquire / complain to (I guess) https://wint.global/kontakt/ about the issue, it is just less likely that they'd feel compelled to fix it)

**Comment by westnordost** [comment]
I'll ask them.

It's confirmed that access to streetcomplete.app is definitely blocked from the hoster side, yes?

**Comment by westnordost** [comment]
It's weird how westnordost.de can return fine and streetcomplete.app not, it's hosted on the same webspace.

Anyway, before sending an email to the hosting provider, I checked it again with https://check-host.net/check-report/2162ee5ek65f and now it seems fine from Russia. Maybe something temporary, then.

Day before yesterday I was also configuring some stuff on the webspace, namely moving all streetcomplete stuff to streetcomplete.app and some really weird things happened then, which amongst other things led to the display of 403 Forbidden even though it shouldn't have and then the whole webspace not being available at all for a short while, as if the server had to restart.

(What I did was to move a directory from westnordost httpdocs folder to streetcomplete httpdocs folder, that led to the above. When I downloaded the folder instead and then uploaded it to the new location, all was fine... 🤷‍♂️).

Anyway, I will file this under "curious issues with the web hoster".

**Comment by deevroman** [comment]
@westnordost You checked http, not httpS. For httpS, the problem is still there

**Comment by westnordost** [comment]
How can I check that? There is no button for that

**Comment by westnordost** [comment]
oh nvm

**Comment by westnordost** [comment]
interestingly, https://www.streetcomplete.app is no problem

**Comment by westnordost** [comment]
Hmm, on my webspace, Apache is used, not ngnix. As far as I know. Maybe those two can be mixed....?

**Comment by westnordost** [comment]
Ha, okay, my guess is that internally, the webhoster uses nginx to manage redirections set in the webhosting management software.

**Comment by mnalis** [comment]
> Ha, okay, my guess is that internally, the webhoster uses nginx to manage redirections set in the webhosting management software.

Yes, I guess they use nginx in reverse proxy mode, so it handles TLS and then forwards a connection to appropriate apache server when needed.

> interestingly, https://www.streetcomplete.app/ is no problem

That should also be explained by above. It looks like it is nginx that generates `301` redirect to `https://streetcomplete.app/`
Only when nginx decides that it needs to forward connection to the apache,  does the `403` error seem to happen.

> westnordost.de also opens in the browser without errors.

That is interesting.  Indeed e.g. `https://westnordost.de/streetcomplete/banned_versions.txt`  loads normally: https://check-host.net/check-report/2164a06dka1a. Different hosts seem to be running with different configs (perhaps inherited at time of creation?), e.g.:

- https://westnordost.de/blog/ seems to be running under `PHP/8.3.14`, _while_
- https://streetcomplete.app/ says it is `X-Powered-By:` older version `PHP/8.1.31`

---

@westnordost Is there an option to choose under which PHP / apache version is site going to be run under (e.g. some separate hoster-provided managment interface? Or some _"upgrade"_ button?)

Another idea:  do you perhaps have some (hidden on UNIXoids) `.htaccess` files in those httpdocs folders? I'm wondering if  adding `Allow` directives from working one (westnordost.de) to broken one (streetcomplete.app) perhaps help? (in case the hoster is adding default deny lists only to those webhosts not specifying their own blacklists)?

**Comment by westnordost** [comment]
**So, some IP range(s) from Russia was blacklisted by the provider for the streetcomplete.app domain.**

It is opaque to me why IP ranges are blocked from the provider side, but I guess this must have been collateral in defense of some cyberattack. It was also by the way not a properly implemented block, as access to www.streetcomplete.app went through just fine, lol.

Anyway, I asked them to remove the block (at least for https://check-host.net, that's the example IP I could give to them) and they did, so I can close the issue.

I expect this to become a recurring issue though. If I wanted to do a cyber attack, I'd make sure to get a VPN to make it appear as if the connection originates from some Russian IP, because given the political situation, I'm pretty sure law enforcement on either side is interested to work together on internet crime. I understand that this can lead to liberal IP-blocks shot from the hip with big collateral.

Now having solved the mystery where this error came from, in the future I will allow my webhosting provider more time before I open a ticket because my expectation is that the block lists they use are limited in time and get removed some time after they were created.

When streetcomplete.app is not reachable, users will not have up-to-date statistics in the app and won't be able to upload photos, otherwise the app should work as normal.
